Submitted By: Ken Moffat <ken@kenmoffat.uklinux.net>
Date: 2005-06-28
Initial Package Version: 0.16.14
Upstream Status: Submitted Upstream
Origin: from ROCK Linux
Description: Allows bin86 to compile on x86_64, which permits lilo to
 be used in a 64-bit system.  I'm dubious about ROCK's attempts to
 automatically add dual-licensing to all their patches, but bin86 is
 already GPL'd.  I've heard from the maintainer that this will be
 included in the next upload.


# --- ROCK-COPYRIGHT-NOTE-BEGIN ---
# 
# This copyright note is auto-generated by ./scripts/Create-CopyPatch.
# Please add additional copyright information _after_ the line containing
# the ROCK-COPYRIGHT-NOTE-END tag. Otherwise it might get removed by
# the ./scripts/Create-CopyPatch script. Do not edit this copyright text!
# 
# ROCK Linux: rock-src/package/x86/bin86/x86_64.patch
# ROCK Linux is Copyright (C) 1998 - 2005 Clifford Wolf
# 
# This patch file is dual-licensed. It is available under the license the
# patched project is licensed under, as long as it is an OpenSource license
# as defined at http://www.opensource.org/ (e.g. BSD, X11) or under the terms
# of the GNU General Public License as published by the Free Software
# Foundation; either version 2 of the License, or (at your option) any later
# version.
# 
# --- ROCK-COPYRIGHT-NOTE-END ---

diff -urN bin86-0.16.14-orig/ld/x86_aout.h bin86-0.16.14/ld/x86_aout.h
--- bin86-0.16.14-orig/ld/x86_aout.h	2004-11-07 20:07:31.506442000 +0100
+++ bin86-0.16.14/ld/x86_aout.h	2004-11-07 20:07:41.106982496 +0100
@@ -12,6 +12,9 @@
 #if defined(i386) || defined(__BCC__) || defined(MSDOS)
 typedef long Long;
 #define __OUT_OK 1
+#elif defined(__x86_64__)
+typedef int Long;
+#define __OUT_OK 1
 #else
 typedef char Long[4];
 #endif
